### PR TITLE
[SERV-1154] Add update-go-version.yml

### PR DESCRIPTION
**Pull request by angelahuqing**

### CODE REVIEW

**Review by DRickard — DISMISSED** [review]

File: .github/workflows/update-go-version.yml [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+          git commit -S -m "Update Go version to $latest_go_version"
+          git push --set-upstream origin $branch_name
+          echo "::set-output name=branch::$branch_name"
+
+      - name: Install GitHub CLI
+        run: sudo apt-get install gh
+  
+      - name: Create pull request
+        if: ${{ steps.check-update.outputs.update_needed == 'true' && steps.check-branch.outputs.branch_not_exist == 'true'}}
+        run: gh pr create -B main -H ${{ steps.update-go-mod.outputs.branch }} --title "Update Go version to ${{ steps.get-latest-go.outputs.latest_go_version }}" --body "This PR updates the Go version in the go.mod file to the latest available version."
```
Comment (ksclarke) [inline .github/workflows/update-go-version.yml:88]: I think using the `|` syntax and `\`s might be a little more readable(?) For instance:

```
run: |
  gh pr create -B main -H ${{ steps.update-go-mod.outputs.branch }} \
    --title "Update Go version to ${{ steps.get-latest-go.outputs.latest_go_version }}" \
    --body "This PR updates the Go version in the go.mod file to the latest available version."
```

Comment (angelahuqing) [reply .github/workflows/update-go-version.yml:88]: I think we should keep it consistent for all of the `run` steps. The other run steps have `|` and don't have `\` but are delineated by a new line. I think a new line is enough to keep it consistent.

Comment (ksclarke) [reply .github/workflows/update-go-version.yml:88]: Isn't there a difference between the earlier command that fits on one line vs. one that is longer and extends over multiple lines? I think the second example needs the `\` to work?

Comment (angelahuqing) [reply .github/workflows/update-go-version.yml:88]: Yes I think you're right. I'll add in the `\`

**Review by ksclarke — APPROVED** [review]

**Review by DRickard — APPROVED** [review]

**Pull request merged**